### DISCUSSION TITLE
Syntax Highlighting in TextView

### DISCUSSION THREAD

**Discussion by edbsystem** [tui-cs/Terminal.Gui#1553] [General]
Loving the framework so far. Just want to ask if I can do syntax highlighting in the TextView?

**Comment by BDisp** [comment]
See here #1336 and the` Syntax Highlighting` scenario on the `UICatalog`.

**Reply by tznind** [reply]
This is the code behind the Syntax Highlighting scenario (see below).  There is no lexer or out of the box parsers so you will have to use a library for that or write your own one.  The way you implement the highlighting is to subclass TextView and override `void ColorNormal (List<System.Rune> line, int idx) `.

https://github.com/migueldeicaza/gui.cs/blob/646a88e4bac25885425bc0382218fd25c56d3157/UICatalog/Scenarios/SyntaxHighlighting.cs#L64-L210

**Reply by edbsystem** [reply]
Thank you so much :-)
